### PR TITLE
[Form] [ChoiceType] Add a `placeholder_attr` option

### PR DESCRIPTION
**Pull request by elementaire**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | #38958
| Similar Pull Request | #47570
| License       | MIT
| Doc PR        | symfony/symfony-docs#17469

Allow to custom the attributes of the placeholder (`data-*`, `title`, `disabled`, etc.).

### CODE REVIEW

**Comment by elementaire** [comment]
Thoughts ?

File: src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig [hunk 1]
```diff
@@ -61,7 +61,7 @@
     {%- endif -%}
     <select {{ block('widget_attributes') }}{% if multiple %} multiple="multiple"{% endif %}>
         {%- if placeholder is not none -%}
-            <option value=""{% if required and value is empty %} selected="selected"{% endif %}>{{ placeholder != '' ? (translation_domain is same as(false) ? placeholder : placeholder|trans({}, translation_domain)) }}</option>
+            <option value=""{% if placeholder_attr is defined and placeholder_attr %}{% with { attr: placeholder_attr } %}{{ block('attributes') }}{% endwith %}{% endif %}{% if required and value is empty %} selected="selected"{% endif %}>{{ placeholder != '' ? (translation_domain is same as(false) ? placeholder : placeholder|trans({}, translation_domain)) }}</option>
```
Comment (xabbuh) [inline src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig]: maybe shorten this a bit:

```suggestion
            <option value=""{% if placeholder_attr|default({}) %}{% with { attr: placeholder_attr } %}{{ block('attributes') }}{% endwith %}{% endif %}{% if required and value is empty %} selected="selected"{% endif %}>{{ placeholder != '' ? (translation_domain is same as(false) ? placeholder : placeholder|trans({}, translation_domain)) }}</option>
```

File: src/Symfony/Bridge/Twig/Resources/views/Form/foundation_5_layout.html.twig [hunk 2]
```diff
@@ -156,7 +156,7 @@
     {%- endif -%}
     <select {{ block('widget_attributes') }}{% if multiple %} multiple="multiple" data-customforms="disabled"{% endif %}>
         {% if placeholder is not none -%}
-            <option value=""{% if required and value is empty %} selected="selected"{% endif %}>{{ translation_domain is same as(false) ? placeholder : placeholder|trans({}, translation_domain) }}</option>
+            <option value=""{% if placeholder_attr is defined and placeholder_attr %}{% with { attr: placeholder_attr } %}{{ block('attributes') }}{% endwith %}{% endif %}{% if required and value is empty %} selected="selected"{% endif %}>{{ translation_domain is same as(false) ? placeholder : placeholder|trans({}, translation_domain) }}</option>
```
Comment (xabbuh) [inline src/Symfony/Bridge/Twig/Resources/views/Form/foundation_5_layout.html.twig]: same here:

```suggestion
            <option value=""{% if placeholder_attr|default({}) %}{% with { attr: placeholder_attr } %}{{ block('attributes') }}{% endwith %}{% endif %}{% if required and value is empty %} selected="selected"{% endif %}>{{ translation_domain is same as(false) ? placeholder : placeholder|trans({}, translation_domain) }}</option>
```

File: src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig (hunk 1 above)
Comment (elementaire) [reply src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig]: Good idea :)

**Review by xabbuh — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @elementaire.

**Comment by Seb33300** [comment]
This may have some relationship to https://github.com/symfony/symfony/issues/45727

**Pull request merged**